src/engine.rs

```diff
@@ -434,6 +439,13 @@ impl Reedline {
         self
     }
 
+    /// Set a different prompt to be used after submitting each line
+    #[must_use]
+    pub fn with_transient_prompt(mut self, transient_prompt: Box<dyn Prompt>) -> Self {
+        self.transient_prompt = Some(Arc::from(Mutex::from(transient_prompt)));
+        self
+    }
```

Comment:
If you can get by with this `fn (mut self, Box<>) -> Self` signature, then I don't think the whole `Arc<Mutex<Box<>>>` is needed, right?

Reply:
@ysthakur sorry for not getting back to you!

Yes I like the variant with `.take()`. To me that is pretty idiomatic.